src/Parser/DeathKnight/Blood/Modules/Features/MarrowrendUsage.js

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import SPELLS from 'common/SPELLS';
+import AbilityTracker from 'Parser/Core/Modules/AbilityTracker';
+import SpellIcon from 'common/SpellIcon';
+import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+
+class MarrowrendUsage extends Analyzer {
+
+  static dependencies = {
+    abilityTracker: AbilityTracker,
+    combatants: Combatants,
+  };
+
+  /*
+    currentBoneShieldBuffer contains the BS stacks caused by the actual MR cast
+    since the applyBuffStack event happens before the cast event
+  */
+
+  currentBoneShieldStacks = 0;
+  currentBoneShieldBuffer = 0;
+  lastMarrowrendCast = 0;
+
+  bsStacksWasted = 0;
+  badMRCasts = 0;
+  refreshMRCasts = 0;
+  totalMRCasts = 0;
+
+  REFRESH_AT_STACKS = 6;
+  REFRESH_AT_SECONDS = 6;
+  BS_DURATION = 30;
+  MR_GAIN = 3;
+
+
+  on_toPlayer_applybuff(event) {
+    if (event.ability.guid === SPELLS.BONE_SHIELD.id){
+      this.currentBoneShieldBuffer += 1;
+      this.currentBoneShieldStacks += 1;
+    }
+  }
+
+  on_toPlayer_applybuffstack(event) {
+    if (event.ability.guid === SPELLS.BONE_SHIELD.id){
+      this.currentBoneShieldBuffer += 1;
+      this.currentBoneShieldStacks += 1;
+    }
+  }
+
+
+  on_toPlayer_removebuff(event) {
+    if (event.ability.guid === SPELLS.BONE_SHIELD.id){
+      this.currentBoneShieldStacks = 0;
+    }
+  }
+
+  on_toPlayer_removebuffstack(event) {
+    if (event.ability.guid === SPELLS.BONE_SHIELD.id){
+      this.currentBoneShieldBuffer = 0;
+      this.currentBoneShieldStacks -= 1;
+    }
+  }
+
+  on_byPlayer_cast(event) {
+    if (event.ability.guid === SPELLS.MARROWREND.id) {
```

Comment:
It's preferred that you avoid nested if statements when possible. You can instead check for the opposite and return early.